### PR TITLE
nift 1.17 (new formula)

### PR DESCRIPTION
**Pull request by n-ham**
- [y] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [y] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [y] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [y] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [y] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Is there any functionality that could be tested?

**Comment by n-ham** [comment]
We could try testing any of the commands on https://www.nift.cc/docs/nift_commands.html 

However without testing them extensively it's unlikely to be much better than checking that the program is indeed installed and runs without any commands given. Thoughts?

**Comment by SMillerDev** [comment]
I'd say building a page would be a better test. Since ideally you'd verify that the program does what the user wants to use it for.

**Comment by n-ham** [comment]
I've almost got a test written to initialise a site and check that the index page was actually built/created, the only problem left is that `assert_predicate testpath/"index.html", :exist?` is saying false is returned even though if I run `system "ls"` index.html is indeed there. Any idea what might be going wrong?

**Comment by SMillerDev** [comment]
Are you actually inside testpath there? Can you force it writing to a specific path?

**Comment by n-ham** [comment]
I make a directory test and cd in to it. Then I run `nsm init` which should create site/index.html, so I cd in to site and try running `assert_predicate testpath/"index.html", :exist?`, should that be something else? I figured testpath is the name of the predicate?

**Comment by n-ham** [comment]
Yep, doing `assert_predicate testpath/"test/site/index.html", :exist?` works. Cheers

**Comment by n-ham** [comment]
Hopefully that's more to your liking :+1:

**Comment by SMillerDev** [comment]
It's not about what I like, it's about the most useful things for homebrew users. And this seems to do well in that regard.

**Comment by n-ham** [comment]
Sorry wasn't trying to be a smartass or anything like that or even directing it to you specifically, more that hopefully it's more to homebrew's liking (as in a more appropriate test) :)

File: Formula/nift.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  sha256 "cdcc3b54d9d98c70a95e14396d09588017fdb2ecfcc821a96778251577b9bafe"
+
+  def install
+    system "make"
+    system "make", "install", "PREFIX=#{prefix}"
+  end
+
+  test do
+    Dir.mkdir "test"
+    Dir.chdir "test"
```
Comment (fxcoudert) [inline Formula/nift.rb]: Unnecessary. The test block already runs in a temporary directory

Comment (n-ham) [reply Formula/nift.rb]: `nsm init` needs to be run in an empty directory or an empty repo

Comment (n-ham) [reply Formula/nift.rb]: The changing directory to the parent at the end might be unnecessarily? I figured I'd leave it back where it started..

**Comment by n-ham** [comment]
I'd be happy to help either of you set up a nice personal site using Nift and an HTML5 UP template if you're interested too. The Nift site (https://nift.cc) is using the escape-velocity template, https://n-ham.com is using the solid-state template.  Feel free to email me (contact@n-ham.com)

File: Formula/nift.rb (hunk 1 above)
Comment (fxcoudert) [reply Formula/nift.rb]: Then:

```
mkdir "empty" do
  system "#{bin}/nsm", "init"
  assert_predicate testpath/"empty/site/index.html", :exist?
end
```

Comment (n-ham) [reply Formula/nift.rb]: Made the changes, tested them (including audit) and have pushed the changes. Let me know if I've missed anything.

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
Thanks @n-ham for your contribution to Homebrew!

**Pull request closed without merging**